cli/command/container/create.go

```diff
@@ -225,26 +217,15 @@ func createContainer(ctx context.Context, dockerCli command.Cli, containerCfg *c
 	if err != nil {
 		return "", err
 	}
+	var namedRef reference.Named
 	if named, ok := ref.(reference.Named); ok {
 		namedRef = reference.TagNameOnly(named)
-
-		if taggedRef, ok := namedRef.(reference.NamedTagged); ok && !options.untrusted {
-			var err error
-			trustedRef, err = image.TrustedReference(ctx, dockerCli, taggedRef)
-			if err != nil {
-				return "", err
-			}
-			config.Image = reference.FamiliarString(trustedRef)
-		}
 	}
 
 	pullAndTagImage := func() error {
 		if err := pullImage(ctx, dockerCli, config.Image, options); err != nil {
 			return err
```

Comment:
After this, we can probably inline this `pullImage` now, and get rid of this closure.

Reply:
`pullAndTagImage` is used twice below - would you rather inline `pullImage` or ditch `pullAndTagImage` and use `pullImage` twice?  (the latter sounds cleaner IMO)

Reply:
(I just pushed the latter in https://github.com/docker/cli/compare/81e57b2d5fc1a95aef77545e182e466aa97c25ce..a5312466d6359558cee6bba555bcbc7e592fb149, but of course I'm happy to adjust further)